pkg/awsutils/awsutils.go

```diff
@@ -1021,6 +1061,34 @@ func (cache *EC2InstanceMetadataCache) GetIPv4PrefixesFromEC2(eniID string) (add
 	return returnedENI.Ipv4Prefixes, nil
 }
 
+// GetIPv6PrefixesFromEC2 calls EC2 and returns a list of all addresses on the ENI
+func (cache *EC2InstanceMetadataCache) GetIPv6PrefixesFromEC2(eniID string) (addrList []*ec2.Ipv6PrefixSpecification, err error) {
+	eniIds := make([]*string, 0)
```

Comment:
nit: this just causes an unnecessary heap allocation.

In general, don't ever do `make([]foo, 0)`.  Instead, just use the nil value.  `nil` is equivalent (but not the same data structure in memory) to an empty list in golang, and this way we delay allocation until we actually add an element.  In the common case of returning an error before actually appending an element, then there's never a heap allocation at all.

```suggestion
	var eniIds []*string
```

In this case, we _know_ we're creating a single element array (see next line of code), so we could also just do that:
```golang
	eniIds := []*string{aws.String(eniID)}
```

Reply:
Makes sense. Will address it.